GitUI/CommandsDialogs/FormRemotes.cs

```diff
@@ -281,51 +310,64 @@ private void btnToggleState_Click(object sender, EventArgs e)
 
         private void SaveClick(object sender, EventArgs e)
         {
-            if (string.IsNullOrEmpty(RemoteName.Text))
+            if (string.IsNullOrWhiteSpace(RemoteName.Text))
             {
                 return;
             }
 
+            var remote = RemoteName.Text.Trim();
+            var remoteUrl = Url.Text.Trim();
+            var remotePushUrl = comboBoxPushUrl.Text.Trim();
             try
             {
                 // disable the control while saving
                 tabControl1.Enabled = false;
 
-                if ((string.IsNullOrEmpty(comboBoxPushUrl.Text) && checkBoxSepPushUrl.Checked) ||
-                    (comboBoxPushUrl.Text.Equals(Url.Text, StringComparison.OrdinalIgnoreCase)))
+                if (string.IsNullOrEmpty(remotePushUrl) && checkBoxSepPushUrl.Checked ||
+                    remotePushUrl.Equals(remoteUrl, StringComparison.OrdinalIgnoreCase))
                 {
                     checkBoxSepPushUrl.Checked = false;
                 }
 
                 // update all other remote properties
                 var result = _remoteManager.SaveRemote(_selectedRemote,
-                                                             RemoteName.Text,
-                                                             Url.Text,
-                                                             checkBoxSepPushUrl.Checked ? comboBoxPushUrl.Text : null,
-                                                             PuttySshKey.Text);
+                                                       remote,
+                                                       remoteUrl,
+                                                       checkBoxSepPushUrl.Checked ? remotePushUrl : null,
+                                                       PuttySshKey.Text);
 
                 if (!string.IsNullOrEmpty(result.UserMessage))
                 {
                     MessageBox.Show(this, result.UserMessage, _gitMessage.Text);
                 }
+                else
+                {
+                    var remotes = Repositories.RemoteRepositoryHistory.Repositories;
+                    RemoteUpdate(remotes, _selectedRemote.Url, remoteUrl);
```

Comment:
I got an exception when adding a repo here, _selectedRemote was null. Not investigated

Reply:
I obviously didn't try to add a new remote.
Should be fixed now - tried all of them: add new, rename, delete

Reply:
Yes, seem to be working now.